aws/data_source_aws_waf_web_acl_test.go

```diff
@@ -0,0 +1,78 @@
+package aws
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+)
+
+func TestAccDataSourceAwsWafRegionalWebAcl_Basic(t *testing.T) {
+	name := "tf-acc-test"
+	resourceName := "aws_wafregional_web_acl.web_acl"
+	datasourceName := "data.aws_wafregional_web_acl.web_acl"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccDataSourceAwsWafRegionalWebAclConfig_NonExistent,
+				ExpectError: regexp.MustCompile(`web ACLs not found`),
+			},
+			{
+				Config: testAccDataSourceAwsWafRegionalWebAclConfig_Name(name),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrPair(datasourceName, "id", resourceName, "id"),
+					resource.TestCheckResourceAttrPair(datasourceName, "name", resourceName, "name"),
+				),
+			},
+		},
+	})
+}
+
+func testAccDataSourceAwsWafRegionalWebAclConfig_Name(name string) string {
```

Comment:
Since we do not need to verify the rules within the Web ACL and they are optional in the resource, can you please remove the extra infrastructure from the test configuration? Thanks!

```go
func testAccDataSourceAwsWafRegionalWebAclConfig_Name(name string) string {
	return fmt.Sprintf(`
resource "aws_wafregional_web_acl" "web_acl" {
  name        = %[1]q
  metric_name = "tfWebACL"

  default_action {
    type = "ALLOW"
  }
}

data "aws_wafregional_web_acl" "web_acl" {
  name = "${aws_wafregional_web_acl.web_acl.name}"
}
`, name)
}
```